cmd/src/debug_common.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/sourcegraph/src-cli/internal/exec"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+/*
+General Stuff
+TODO: file issue on the existence of OAuth signKey which needs to be redacted
+TODO: Create getSiteConfig function
+*/
+
+type archiveFile struct {
+	name string
+	data []byte
+	err  error
+}
+
+func archiveFileFromCommand(ctx context.Context, path, cmd string, args ...string) *archiveFile {
+	f := &archiveFile{name: path}
+	f.data, f.err = exec.CommandContext(ctx, cmd, args...).CombinedOutput()
+	if f.err != nil {
+		f.err = errors.Wrapf(f.err, "executing command: %s %s: received error: %s", cmd, strings.Join(args, " "), f.data)
+	}
+	return f
+}
+
+// This function prompts the user to confirm they want to run the command
+func verify(confirmationText string) (bool, error) {
+	input := ""
+	for strings.ToLower(input) != "y" && strings.ToLower(input) != "n" {
+		fmt.Printf("%s [y/N]: ", confirmationText)
+		if _, err := fmt.Scanln(&input); err != nil {
+			return false, err
+		}
+	}
+
+	return strings.ToLower(input) == "y", nil
+}
+
+// setOpenFileLimits increases the limit of open files to the given number. This is needed
+// when doings lots of concurrent network requests which establish open sockets.
+func setOpenFileLimits(n uint64) error {
+
+	var rlimit syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit)
+	if err != nil {
+		return err
+	}
+
+	rlimit.Max = n
+	rlimit.Cur = n
+
+	return syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rlimit)
+}
+
+// write to archive all the outputs from kubectl call functions passed to buffer channel
```

Comment:
```suggestion
// write all the outputs from an archive command passed on the channel to to the zip writer
```

(We don't know it's buffered here)